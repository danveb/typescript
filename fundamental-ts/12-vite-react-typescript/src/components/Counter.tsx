import { ReactNode } from "react";

type CounterProps = {
    setCount: React.Dispatch<React.SetStateAction<number>>, 
    children: ReactNode, 
}; 

// pass CounterProps for our props
export default function Counter({ setCount, children }: CounterProps) {
    // useState; start counter number at 1
    // const [count, setCount] = useState<number>(1); 

    return (
        <>
            <h1>{children}</h1>
            <button onClick={() => setCount(prev => prev + 1)}>+</button>
            <button onClick={() => setCount(prev => prev - 1)}>-</button>
        </>
    )
}