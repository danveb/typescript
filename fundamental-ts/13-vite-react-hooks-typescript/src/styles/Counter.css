.card {
    padding: 1.5em;
}

.card button {
    border-radius: 8px;
    border: 1px solid transparent;
    padding: 0.6em 1.2em;
    font-size: 1em;
    font-weight: 500;
    font-family: inherit;
    background-color: #1a1a1a;
    color: #fff; 
    cursor: pointer;
    transition: border-color 0.25s;
}
.card button:hover {
    border-color: #646cff;
}
.card button:focus,
.card button:focus-visible {
    outline: 4px auto -webkit-focus-ring-color;
}

.card p {
    display: inline-block; 
    padding: 0 25px; 
}