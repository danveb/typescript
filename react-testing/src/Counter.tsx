import { useState } from "react";

export const Counter: React.FC = () => {
  const [count, setCount] = useState<number>(0); 

  return (
    <div>
      <button onClick={() => setCount((count) => count + 1)}>Add One</button>
      <div role="contentinfo">Count is {count}</div>
    </div>
  )
}