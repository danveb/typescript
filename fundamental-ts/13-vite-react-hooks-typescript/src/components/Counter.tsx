import { useState } from "react"; 
import "../styles/Counter.css"; 

// create a counter with buttons
// button + to increase count -> onClick 
// button - to decrease count -> onClick 
// if count is below a certain threshold we reset back to 0

export default function Counter() {
    // useState 
    // we can be explicit and define the type count/setCount is 
    // define as number starting at 0
    const [count, setCount] = useState<number>(0); 

    // handleClickUp
    const handleClickUp = (): void => {
        setCount((prevCount) => prevCount + 1); 
    };
    // handleClickDown
    const handleClickDown = (): void => {
        setCount((prevCount) => prevCount - 1); 
    };

    return (
        <div className="card">
            <button onClick={handleClickUp}>+</button>
            <p>Count is {count}</p>
            <button onClick={handleClickDown}>-</button>
        </div>
    )
}