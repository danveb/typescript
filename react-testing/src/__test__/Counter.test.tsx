import { render, screen, fireEvent } from "@testing-library/react";
import { Counter } from "../Counter";

describe("test Counter component", () => {
  // smoke test
  test("renders without crashing", () => {
    render(<Counter />); 
  });

  // snapshot test
  test("creates a snapshot", () => {
    const { asFragment } = render(
      <Counter />
    ); 
    expect(asFragment()).toMatchSnapshot(); 
  }); 

  test("handles a total count of 2 when clicked", () => {
    render(<Counter />); 
    const divElement = screen.getByRole("contentinfo"); 
    const buttonElement = screen.getByText("Add One"); 
    fireEvent.click(buttonElement); 
    fireEvent.click(buttonElement); 
    expect(divElement).toHaveTextContent("Count is 2"); 
  }); 
});