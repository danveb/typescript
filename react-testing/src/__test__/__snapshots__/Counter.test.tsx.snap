// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Counter component creates a snapshot 1`] = `
<DocumentFragment>
  <div>
    <button>
      Add One
    </button>
    <div
      role="contentinfo"
    >
      Count is 0
    </div>
  </div>
</DocumentFragment>
`;
